### PR TITLE
Update DG Design section and Implementation for done command

### LINKED ISSUE #47: Update Developer Guide
On Architecture Diagrams and EditCommands implementations.

### PR DESCRIPTION
**Pull request by AhQuanz**
[ModelClassDiagram](https://drive.google.com/file/d/1nc1nCg93KOPD7zl4yyL5aHHvNfW8SCuq/view?usp=sharing)
[CustomerClassDiagram](https://drive.google.com/file/d/1FPDCpq2p0v-RbbL4-MU2ZgyyLFnOgIr7/view?usp=sharing)
[OrderClassDiagram](https://drive.google.com/file/d/1hWltzoyqTHcd10mi3DfmPRZEDmOixV_A/view?usp=sharing)
[CheeseClassDiagram](https://drive.google.com/file/d/1eMxOhHtsb7fWAzrW3JNNVZatfE1fuajA/view?usp=sharing)
[DoneCommandSeqDiagram](https://drive.google.com/file/d/16kVlnGlLxxruBIrUYscvUEG5ufVnYx6h/view?usp=sharing)

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -23,12 +23,66 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
...
+
+Step 4. `DoneCommand.execute()` will call `DoneCommand.createDoneOrder()` to create a new `order`
+if there is enough unassigned Cheese(s) from `ModelManager.getUnassignedCheeses()`
+
+Step 5. `DoneCommand.execute()` will call `ModelManager.setOrder()` 
+to replace the original order with the new order created from `DoneCommand.createDoneOrder()` 
+and calls `ModelManager.updateCheesesStatus()` to update all cheese(s)'s assign status used for this order.
+
+The following sequence diagram shows hwo the operation `done 1` is carried out as detailed above.
+
```
Comment (weixue123) [inline docs/DeveloperGuide.md:334]: Is there supposed to be a sequence diagram here?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -23,12 +23,66 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
...
+
+#### Design consideration 
+* Aspect : Searching for available cheese(s) for the order
+    * Current choice : Searching for unassigned cheeses for the order is implemented in the `Model.AddressBook`.
+        * Pros: no dependency between `Done` command and `Cheese`,
+        * Cons: performance issues due to multiple functions calls.
+    * Alternative 1 : `Done` command will search for unassigned cheeses.
+        * Pros: better in terms of performance as there are lesser functions to be called .
```
Comment (weixue123) [inline docs/DeveloperGuide.md]: Do clean up the documentation a bit (typos, punctuations, etc.)

Also, take note of the trailing white spaces:
Screen Shot 2021-03-31 at 10 12 59 AM

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -23,12 +23,66 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
 
 ### Model component
 
+![Structure of the Model Component](images/ModelClassDiagram.png)
+
+![Structure of the Customer Component](images/CustomerClassDiagram.png)
+
+![Structure of the Cheese Component](images/CheeseClassDiagram.png)
+
+![Structure of the Order Component](images/OrderClassDiagram.png)
```
Comment (weixue123) [inline docs/DeveloperGuide.md:109]: Class diagrams LGTM

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (AhQuanz) [reply docs/DeveloperGuide.md:334]: Yeap, will be updating in the next commit

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (AhQuanz) [reply docs/DeveloperGuide.md]: Sure. will fix it in the next commit.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -88,6 +105,44 @@ To better illustrate the idea - take, for instance, when the user inputs `listch
 
 --------------------------------------------------------------------------------------------------------------------
 
+### Mark an order as complete feature
+
+#### implementation
+
+Marking an order as completed is implemented in [`DoneCommand.java`](https://github.com/AY2021S2-CS2103-W16-2/tp/blob/master/src/main/java/seedu/address/comamnds/DoneCommand.java).
```
Comment (laurenlhy) [inline docs/DeveloperGuide.md]: This link doesn't work, there is also a typo in "commands"
I think it should be address/logic/commands/DoneCommand.java?

**Comment by daniellau88** [comment]
hihi, do resolve the comments in gdrive too

**Comment by AhQuanz** [comment]
> hihi, do resolve the comments in gdrive too

Where are the comments?

**Review by daniellau88 — APPROVED** [review]
LGTM, can merge in once u done changing :)

**Pull request merged**